sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubJsonClient.java

```diff
@@ -136,20 +136,23 @@ public int publish(TopicPath topic, List<OutgoingMessage> outgoingMessages)
     for (OutgoingMessage outgoingMessage : outgoingMessages) {
       PubsubMessage pubsubMessage = new PubsubMessage().encodeData(outgoingMessage.elementBytes);
 
-      Map<String, String> attributes = outgoingMessage.attributes;
-      if ((timestampAttribute != null || idAttribute != null) && attributes == null) {
-        attributes = new TreeMap<>();
+      Map<String, String> attributes = null;
+      if ((timestampAttribute != null || idAttribute != null)) {
+        if (outgoingMessage.attributes == null) {
```

Comment:
Is it legal for this to be null? It's not marked as @Nullable

Reply:
Yeah, it can be null. The PubsubMessage used here isn't org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage, it's com.google.api.services.pubsub.model.PubsubMessage. See https://developers.google.com/resources/api-libraries/documentation/pubsub/v1/java/latest/com/google/api/services/pubsub/model/PubsubMessage.html#getAttributes()